kroxylicious/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -65,6 +79,13 @@ public KafkaProxy(Configuration config) {
         this.useIoUring = config.proxy().useIoUring();
 
         this.filterChainFactory = new FilterChainFactory(config);
+
+        if (config.proxy().keyStoreFile() != null) {
```

Comment:
I'm not personally a fan of `null` guards, I think I'd prefer `keyStoreFile` to return an optional or the more invasive option of pushing the `SslContext` creation into `proxyConfig` and hide all construction logic in there.